### PR TITLE
Add configuration for stale issue bot

### LINKED ISSUE #228: Configure a bot that closes stale issues/PRs
What do we think about having a [bot](https://github.com/apps/stale) that automatically marks/closes stale issues?

### PR DESCRIPTION
**Pull request by andersy005**
- [ ] Closes #228

@jbusecke, this is my attempt at addressing #228. I haven't added configurations for PRs because I wasn't sure whether you want to close stale PRs. If this is the case, let me know and I will add the necessary configurations. Also, GitHub should have submitted a request to the xgcm organization on my behalf. This request is about installing the https://github.com/probot/stale GitHub on this repo.

### CODE REVIEW

File: .github/stale.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# Number of days of inactivity before an Issue or Pull Request with the stale label is closed.
+# Set to false to disable. If disabled, issues still need to be closed manually, but will remain marked as stale.
+daysUntilClose: 30
+
+# Only issues or pull requests with all of these labels are check if stale. Defaults to `[]` (disabled)
+onlyLabels: []
+
+# Issues or Pull Requests with these labels will never be considered stale. Set to `[]` to disable
+exemptLabels: []
```
Comment (andersy005) [inline .github/stale.yml:14]: >  Is there a way to mark issues that you want to keep open? 

We could use this list of exempt labels.

**Comment by andersy005** [comment]
>  Also, GitHub should have submitted a request to the xgcm organization on my behalf. This request is about installing the https://github.com/probot/stale GitHub on this repo. 

I switched to a GitHub based workflow. So, this isn't needed anymore

File: .github/stale.yml (hunk 1 above)
Comment (andersy005) [reply .github/stale.yml]: Done! Let me know if you still need us to mark PRs as stale

**Comment by jbusecke** [comment]
Should we close this? Sorry if I misunderstood.

**Comment by andersy005** [comment]
> Should we close this? Sorry if I misunderstood.

Sorry for the confusion. I meant that we don't need to use https://github.com/probot/stale bot. We can use GitHub action directly. Let's merge this if the configuration looks right to you.

**Review by jbusecke — APPROVED** [review]
This looks great. Ill merge once the checks pass.

**Pull request merged**